pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1047,57 +988,23 @@ async def parse_packages_components(self, comps_result):
 
     async def parse_packages(self, primary_xml, filelists_xml, other_xml, file_extension="gz"):
         """Parse packages from the remote repository."""
-        packages = await RpmFirstStage.parse_repodata(
+        parser = MetadataParser.from_metadata_files(
             primary_xml.path, filelists_xml.path, other_xml.path
         )
 
-        # skip SRPM if defined
-        if "srpm" in self.skip_types and not self.mirror:
-            packages = collections.OrderedDict(
-                (pkgId, pkg) for pkgId, pkg in packages.items() if pkg.arch != "src"
-            )
-
         progress_data = {
             "message": "Parsed Packages",
             "code": "sync.parsing.packages",
-            "total": len(packages),
+            "total": parser.count_packages(),
```

Comment:
I think this will count SRPMs as well, even if we skip them.

Reply:
Though we did parse SRPMs anyway, however I'm not sure if it just confuses a user.

Reply:
There's not really a good answer, I agree.  I'm slightly partial to the status quo even if it's slightly confusing to have a mismatch in the numbers parsed vs synced.  If we do it the other way around the confusion would be perhaps less likely, but more severe, if the user expected a certain number of packages in the repo.